syntax = "proto3";

package cameraReaderWindows;

enum CaptureEncoding {

	Any = 0;
	Unknown = 1;
	ARGB = 2;
	XRGB = 3;

	I420 = 4;
	NV12 = 5;
	YV12 = 6;
	Y800 = 7;

	YVYU = 8;
	YUY2 = 9;
	UYVY = 10;
	HDYC = 11;

	MJPEG = 12;
	H264 = 13;
}

message CaptureFormat {
    uint32 width = 1;
    uint32 height = 2;
    uint64 frameinterval = 3;
    CaptureEncoding encoding = 4;
}

message CameraReaderWindows {
    string cameraName = 5;
	string cameraPath = 6;
    repeated CaptureFormat formats = 7;
}

message CameraList {
    repeated CameraReaderWindows cameras = 7;
}

enum Flip {
	FlipNone = 0;
	FlipVertically = 1;
	FlipHorizontally = 2;
	FlipBoth = 3;
}

message StartCaptureArguments {
	string cameraName = 8;
	string cameraPath = 9;
	uint32 width = 10;
	uint32 height = 11;
	uint64 frameinterval = 12;
	CaptureEncoding encoding = 13;
	Flip flippingMode = 14;
}

enum StartResult {
	Success = 0;
	InUse = 1;
	Error = 2;
}

message StartCaptureResult {
	bool canResetGraph = 14;
	bool canSetAudioConfig = 15;
	bool canSetVideoConfig = 16;
	bool canConnectFilters = 17;
	StartResult result = 18;
	uint64 devicePointer = 19;
}